apps/src/blockly/addons/cdoFieldDropdown.js

```diff
@@ -109,54 +112,56 @@ export default class CdoFieldDropdown extends GoogleBlockly.FieldDropdown {
-    // Convert this list to a string of dropdown options separated by commas and assign to options.
-    // Otherwise, assign options to config string.
-    // Note that `config` is either a printer-style number range or a string of options separated
-    // by commas, but not both. For example, a `config` like "1,6-9, &quot;SLOTH&quot;"
-    // would NOT be supported.
-    let options =
-      numberList.length > 0 ? numberListToString(numberList) : config;
-    options = options.split(',').map(val => {
-      val = val.trim();
-      // If val is one of the options in this.menuGenerator_,
-      // human-readable string is displayed.
-      if (originalOptionsMap[val]) {
-        return [originalOptionsMap[val], val];
-      } else {
-        // Remove quotes and display option with lowercase characters.
-        // For example, "GIRAFFE" would be transformed to giraffe.
-        const humanReadableVal = this.toHumanReadableString(val);
-        return [humanReadableVal, val];
-      }
-    });
-    return options;
-  }
+/**
+ * Updates field options based on `config` attribute of field
+ * @param config attribute of field
+ * @return Array of option arrays
+ */
+function getUpdatedOptionsFromConfig(config, menuGenerator) {
+  // If `menuGenerator_` is an array, it is an array of options with
+  // each option represented by an array containing 2 elements -
+  // a human-readable string and a language-neutral string. For example,
+  // [['first item', 'ITEM1'], ['second item', 'ITEM2'], ['third item', 'ITEM3']].
+  // Options are included in the block definition.
+  const originalOptionsMap = Array.isArray(menuGenerator)
+    ? menuGenerator.reduce((optionsMap, curr) => {
+        optionsMap[curr[1]] = curr[0];
+        return optionsMap;
+      }, {})
+    : {};
+  const numberList = printerStyleNumberRangeToList(config);
+  // If numberList is assigned a non-empty array, it contains a list of numbers.
+  // Convert this list to a string of dropdown options separated by commas and assign to options.
+  // Otherwise, assign options to config string.
+  // Note that `config` is either a printer-style number range or a string of options separated
+  // by commas, but not both. For example, a `config` like "1,6-9, &quot;SLOTH&quot;"
+  // would NOT be supported.
+  let options = numberList.length > 0 ? numberListToString(numberList) : config;
+  options = options.split(',').map(val => {
+    val = val.trim();
+    // If val is one of the options in this.menuGenerator_,
+    // human-readable string is displayed.
+    if (originalOptionsMap[val]) {
+      return [originalOptionsMap[val], val];
+    } else {
+      // Remove quotes and display option with lowercase characters.
+      // For example, "GIRAFFE" would be transformed to giraffe.
+      const humanReadableVal = toHumanReadableString(val);
+      return [humanReadableVal, val];
+    }
+  });
+  return options;
 }
+
+export var __TestInterface = {
```

Comment:
I think it would be more straightfoward to just export the function directly, I'm guessing this style in the example you used was copied from somewhere else given the old var syntax.

Reply:
Yeah - I was thinking of just exporting the function, but wrote in my PR description about how it was done for `fixMenuGenerator` [here](https://github.com/code-dot-org/code-dot-org/blob/00eaea1f87e713ffd51e95f1d0c8e51e3506f441/apps/src/blockly/addons/cdoFieldImageDropdown.js#L136) in `CdoFieldImageDropdown.js`. I don't feel strongly either way, but I think being consistent would be good. I can change to just exporting the function being tested in both places. Thoughts?

Reply:
Yeah sorry I meant the example you copied also looks like it was copied from somewhere else. I'd just export the function being tested, but I guess it's just personal preference. There are only a couple examples like that in our code base (a couple more if you use const instead of var):

![image](https://github.com/code-dot-org/code-dot-org/assets/25372625/53c28bf1-9146-4937-a1e3-06862b8fc495)

Reply:
Cool - I'll go ahead and export the functions then instead of using `TestInterface`.